src/main.js

```diff
@@ -1699,12 +1671,58 @@ const DiscoveryFeedApplication = new Lang.Class({
         this.parent(connection, path);
     },
 
+    _createWindowResources: function() {
+        this._window = new DiscoveryFeedMainWindow({
+            application: this,
+            type_hint: !this._debugWindow ? Gdk.WindowTypeHint.DOCK : Gdk.WindowTypeHint.NORMAL,
+            role: !this._debugWindow ? SIDE_COMPONENT_ROLE : null,
+            card_model: this._discoveryFeedCardModel
+        });
+
+        // to be able to set the opacity from css
+        let visual = Gdk.Screen.get_default().get_rgba_visual();
+        if (visual) {
+            this._window.set_visual(visual);
+        }
+
+        // when the label contains letters and numbers the allocation
+        // does not work properly, force re-allocation here
+        this._window.expanded_date.realize();
+
+        this._window.connect('notify::visible', Lang.bind(this, this._onVisibilityChanged));
+
+        this._setupWindowInteraction();
+
+        // update position when workarea changes
+        let display = Gdk.Display.get_default();
+        display.connect('monitor-added', Lang.bind(this,
+                                                   this._updateGeometry));
+        display.connect('monitor-removed', Lang.bind(this,
+                                                     this._updateGeometry));
+        let monitor = display.get_primary_monitor();
+        monitor.connect('notify::workarea', Lang.bind(this,
+                                                      this._updateGeometry));
+        this._updateGeometry();
+
+        // We release the hold that gtk takes over the application so that
+        // our inactivity timeout will still work
+        this.release();
```

Comment:
We could also destroy the window in `hide()` instead of simply hiding it, but I guess this is OK too.